lib/qesapdeployment.pm

```diff
@@ -2311,6 +2312,52 @@ sub qesap_az_create_sas_token {
     return script_output($cmd);
 }
 
+=head2 qesap_az_list_container_files
+
+Returns a list of the files that exist inside a given path in a given container
+in Azure storage.
+
+Generated command looks like this:
+
+az storage blob list 
+--account-name <account_name> 
+--container-name <container_name> 
+--sas-token "<my_token>" 
+--prefix <path_inside_container> 
+--query "[].{name:name}" --output tsv
+
+=over 4
+
+=item B<STORAGE> - Storage account name used fur the --account-name argument in az commands
+
+=item B<CONTAINER> - container name within the storage account
+
+=item B<TOKEN> - name of the SAS token to access the account
+
+=item B<PREFIX> - the local path inside the container (to list file inside a folder named 'dir', this would be 'dir')
+
+=back
+=cut
+
+sub qesap_az_list_container_files {
+    my (%args) = @_;
+    foreach (qw(storage container token prefix)) { croak "Missing mandatory $_ argument" unless $args{$_}; }
+    my $cmd = join(' ',
+        'az storage blob list',
+        '--account-name', $args{storage},
+        '--container-name', $args{container},
+        '--sas-token', $args{token},
+        '--prefix', $args{prefix},
+        '--query "[].{name:name}" --output tsv');
```

Comment:
I do not want you to change anything but recently I became big fun of `--output json` and perl `decode_json` combo :-)

https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/sles4sap/azure_cli.pm

```
use Mojo::JSON qw(decode_json);


my $cmd = join(' ',
       ...
        '--query "[].{name:name}" --output json');
    return decode_json(script_output($cmd));
```

Reply:
I am fan of that as well. :D 
You can even manipulate --query to show you only stuff you need, or format it to some degree.